### ISSUE TITLE
Indicator Applet Crashes the SpiralLinux MATE Panel

### ISSUE THREAD

**Issue by JCSMATE** [SpiralLinux/SpiralLinux-project#468]
![Screenshot at 2024-08-20 11-14-53](https://github.com/user-attachments/assets/ac65096f-fc48-4f03-8b0b-a93dda6b3a1e)
The "Indicator Applet" for the SpiralLinux MATE Panel (as well as a few other applets of type "Indicator") appears to CRASH and FREEZE the MATE Panel. Favorably, I had Plank already set up with the MATE System Monitor and my favorite terminal QTerminal, so I was able to play around and regain control (courtesy of "man mate-panel"). Rebooting did NOT fix it.

It would seem that some of the SpiralLinux MATE applets are not properly implemented. . . .

Acceptable Panel Items That DO Work

- Brisk Menu (applet)
- Show Desktop (applet)
- Screensaver (dragged from the menu)
- Power Management (dragged from the menu)
- Workspace Switcher (applet)
- Window List (applet)
- Notification Area [sic] (applet)
- Disk Mounter (applet)
- Clock (applet)

I tried to install the Indicator Applet because I was considering having the notification history list that (I think) that applet offers. Since this problem has unfolded, however, I've completely stopped wanting a notification history list. . . .

I had to force-quit the mate-panel using the system monitor, before the mate-panel command-line commands were able to function properly. My particular Plank set up saved me by allowing me the needed access. This is a pretty serious bug, since it can make regaining control VERY difficult--depending upon the user's situation. . . .

Other than this one horrific discovery, I'm LOVING SpiralLinux MATE.

**Comment by geckolinux** [comment]
Hi there, unfortunately SpiralLinux has no control over this sort of bug since it is built from unmodified Debian packages. SpiralLinux simply avoids some Debian bugs by not installing certain buggy packages, and since the Indicator Applet isn't installed by default it's not a SpiralLinux configuration issue.

With that being said, the Indicator Applet with the `ayatana-indicator-notifications` applet seems to work for me:

![Screenshot from 2024-08-21 21-00-37](https://github.com/user-attachments/assets/7a698e90-f344-4be3-9552-433b6c4fcc10)

![Screenshot from 2024-08-21 21-02-11](https://github.com/user-attachments/assets/c15e57b1-884c-4a85-ba55-3b9b08c07c9c)


![Screenshot from 2024-08-21 21-01-45](https://github.com/user-attachments/assets/8eeb7f26-0fd2-4777-983e-de060cf511aa)

**Comment by JCSMATE** [comment]
![Screenshot at 2024-08-24 16-39-00](https://github.com/user-attachments/assets/fbae5c53-e420-4a90-9694-9b293e689865)
Thank you very much for the information. It has given me a greater understanding of the design and behavior of the "recent notifications" feature of the MATE desktop.

I installed the ayatana components, including ayatana-settings, and enabled recent notifications at startup. However, since the only way to access the recent notifications feature is via the MATE panel's "Indicator Applet" or "Indicator Applet Complete," and since my experience (which includes Ubuntu MATE) with these applets has been less than favorable, I have decided to relinquish my desire for the recent notifications feature on MATE. Ubuntu MATE was set up in such a way that the inherent danger of the design and behavior of the Indicator Applet was hidden; but, working with Debian has REVEALED the danger and caused me to want to avoid this applet entirely. I much prefer the SpiralLinux MATE default of the "notification area" applet; it's too bad that ayatana's recent notifications won't show up there. . . . I also tried to find the location of an ayatana log file that I might access via a shortcut on either the desktop or the panel, but couldn't find any text-file log. . . .

EVERY system will have its idiosyncrasies and its niceties. MATE's idiosyncrasies can be numerous; but, generally speaking, they are easily managed, so MATE is able to give a good degree of satisfaction. Though it would be nice to have recent notifications, if they were a priority for me, I'd go with Cinnamon, Budgie (?), LXQt (?), or Plasma.

As it currently stands, SpiralLinux MATE is still VERY satisfying.
